pilot/pkg/xds/discovery.go

```diff
@@ -333,6 +334,12 @@ func (s *DiscoveryServer) globalPushContext() *model.PushContext {
 
 // ConfigUpdate implements ConfigUpdater interface, used to request pushes.
 func (s *DiscoveryServer) ConfigUpdate(req *model.PushRequest) {
+	if len(model.ConfigsOfKind(req.ConfigsUpdated, kind.Address)) > 0 {
+		// This is a bit like clearing EDS cache on EndpointShard update. Because Address
+		// types are fetched dynamically, they are not part of the same protections, so we need to clear
+		// the cache.
+		s.Cache.ClearAll()
```

Comment:
Why do you clear cache , seems we donot cache ambient specific resources

Reply:
It impacts EDS